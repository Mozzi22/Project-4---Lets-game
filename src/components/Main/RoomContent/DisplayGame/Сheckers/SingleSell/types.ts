export type TSingleSell = {
    id: number | string;
}