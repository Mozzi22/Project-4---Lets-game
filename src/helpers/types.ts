export type TRegBody = { login: string, password: string, confirm: string };
export type TLoginBody = { login: string, password: string };
