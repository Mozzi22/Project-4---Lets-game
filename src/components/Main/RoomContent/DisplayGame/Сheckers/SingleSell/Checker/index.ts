// import { connect } from 'react-redux';
// import { getCurrentChecker } from 'src/Store/games/selectors';
// import { setCurrentChecker } from 'src/Store/games/actions';
// import { ApplicationState } from 'src/Store/types';
import Checker from './Checker';
// import { Dispatch } from 'redux';
// import { TCurrentChecker } from 'src/Store/games/types';

// const mapStateToProps = (state: ApplicationState) => ({
//   currentChecker: getCurrentChecker(state),
// });
// const mapDispatchToProps = (dispatch: Dispatch) => ({
//   setCurrentChecker: (payload) => dispatch(setCurrentChecker(payload)),
// })

export default Checker;
