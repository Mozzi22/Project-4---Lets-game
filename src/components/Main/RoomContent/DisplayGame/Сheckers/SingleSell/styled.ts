import styled from 'styled-components';

export const StSingleSell = styled.div <{ backgroungColor?: string }>`
    width: 50px;
    height: 50px;
    cursor: pointer;
    background-color: ${({ backgroungColor }) => backgroungColor};
`;
