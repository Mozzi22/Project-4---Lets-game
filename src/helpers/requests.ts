import { routes } from '../constants/routes';
import { TUrl } from './types';

const { baseUrl } = routes;

export const getRequest = async (url: TUrl) => {
  const options = { method: 'GET' };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const postRequest = async (url: TUrl, body) => {
  const options = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};
