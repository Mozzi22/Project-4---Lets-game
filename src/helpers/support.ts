export const support = {
    setSessionStorageItem: (name: string, data: string) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        localStorage.setItem(name, data);
    },
    killSessionStorageItem: (name: string) => {
        localStorage.removeItem(name);
    },
    getSessionStorageItem: (name: string) => {
        const item = localStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
};
