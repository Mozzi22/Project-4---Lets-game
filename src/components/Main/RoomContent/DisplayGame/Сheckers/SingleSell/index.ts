import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import SingleSell from './SingleSell';
import { getTicStatus, getPossibleStepPosition } from 'src/Store/games/selectors';
import { ApplicationState } from 'src/Store/types';
import { getPosibleStep, doCheckerStep } from 'src/Store/games/actions';

const mapStateToProps = (store: ApplicationState, props) => ({
    status: getTicStatus(store, props.id),
    position: getPossibleStepPosition(store, props.id),
});
const mapDispatchToProps = (dispatch: Dispatch) => ({
    getPosibleStep: (payload: string) => dispatch(getPosibleStep(payload)),
    doCheckerStep: (payload: string) => dispatch(doCheckerStep(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SingleSell);
