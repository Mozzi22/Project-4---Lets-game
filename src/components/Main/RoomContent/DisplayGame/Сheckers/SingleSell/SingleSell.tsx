import React from 'react';
import { StSingleSell } from './styled';

const SingleSell = ({ id }) => {
    const handleClick = (e) => {
        console.log(e.target.id);
    };

    return (
        <StSingleSell id={id + 1} style={{ background: parseInt((id / 8) + id) % 2 === 0 ? 'black' : 'white' }} onClick={handleClick}>{id + 1}</StSingleSell>
    );
};

export default SingleSell;
