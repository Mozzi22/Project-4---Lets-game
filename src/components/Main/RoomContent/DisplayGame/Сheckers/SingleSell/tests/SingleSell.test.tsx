import React from 'react';
import SingleSell from '../SingleSell';
import { mount, shallow } from 'enzyme';
import { shallowSmart, mountSmart } from 'src/helpers/testHelper';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const store = mockStore({
    game: {
      currentChecker: {}
  },
});

describe('SingleSell', () => {
    const props = {
        id: '',
        position: 1,
        isChecked: false,
        color: 'red',
    }
    it('Should match snapshot', () => {
        const component = shallow(<SingleSell {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StSingleSell', () => {
        const component = mount(<SingleSell {...props}/>);
        expect(component.find('styled__StSingleSell')).toHaveLength(1);
    });
    it('should not render Checker', () => {
        const props = {
            id: 1,
            isChecked: false, 
            color: 'red',
        }
        const component = mount(<SingleSell {...props}/>);
        expect(component.find('Checker')).toHaveLength(0);
    });
       it('should render Checker', () => {
        const props = {
            id: 1,
            isChecked: true, 
            color: 'red',
        }
           const component = mountSmart(<SingleSell {...props} />, store);
           console.log(component.debug());
        expect(component.find('Checker')).toHaveLength(1);
    });
});
