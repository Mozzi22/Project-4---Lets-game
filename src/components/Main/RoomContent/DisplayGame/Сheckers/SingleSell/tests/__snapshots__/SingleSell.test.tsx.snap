// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSell Should match snapshot 1`] = `"<div id=\\"1\\" class=\\"styled__StSingleSell-sc-1cetqro-0 gxBDod\\">1</div>"`;
